### ISSUE TITLE
KfDef spec v1beta1; remove ksonnet nomenclature - make it more application centric

### ISSUE THREAD

**Issue by jlewi** [kubeflow/kubeflow#3478]
KfDef is currently defined here:
https://github.com/kubeflow/kubeflow/blob/5f1762a5a973d3441f010096357ba882f2c9386e/bootstrap/pkg/apis/apps/kfdef/v1alpha1/application_types.go#L126

In 0.7 we should probably create a new version of this structure. 

The current version of this structure is very ksonnet centric; e.g. components & packages are ksonnet concepts. 

Kubeflow is using "Applications" to refer to the different parts of Kubeflow so we should think about adopting that terminology.

kfctl should provide backwards compatibility with the current version at least for 0.7.

**Comment by jlewi** [comment]
I think we are in good position to go ahead and define a go package defining just the v1beta1 spec.
i.e. a package similar to 

The [design doc](https://docs.google.com/document/d/1kviNOdjoBPeoJ56KBvMKWq7SzrSN5rFUQoULK8muD-M/edit?usp=sharing)  and kubeflow/kubeflow/issues/3709
provide a pretty clear idea of what fields should be removed from the v1alpha1 spec.

https://github.com/kubeflow/kubeflow/tree/master/bootstrap/pkg/apis/apps/kfdef/v1alpha1

We need that package to unblock supporting v1beta1 in kfctl.

Bumping to P0 because I think we need to get this done this week or next to keep kfctl on track for 0.7

/cc @yanniszark @gabrielwen @kkasravi

**Comment by jlewi** [comment]
/assign @gabrielwen 

@gabrielwen we can close this as soon as your PR defining the v1beta1 spec is merged.

**Comment by jlewi** [comment]
Duplicate of #4002
